### ISSUE TITLE
Proportional-Width fonts for <code> sections.

### ISSUE THREAD

**Issue by kentfredric** [preaction/Statocles#420]
There's a really unflattering bug that occurs because something about linux+firefox+fontconfig is a hate machine.

You can see here, my code is shown using a proportional-width font.

![snapshot93](https://cloud.githubusercontent.com/assets/44790/12031719/75639884-ae74-11e5-8965-36ec4600b984.png)

The firefox "Fonts" panel indicates the Font Employed is ... Times New Roman. 

It appears like the `font-family` field in normalize.css is not interpreted as a list of fonts, but somehow a lengthy font delimited by commas.

Replacing the `font-family` field with a string of garbage that obviously cant match a font ( eg: "asdrftgh" ) results in the exact same rendering result.

But given the font in question is written:
```css
font-family: monospace, monospace;
```

I can just pragmatically delete one of those two, and it then starts rendering as Courier New.

![snapshot94](https://cloud.githubusercontent.com/assets/44790/12031793/7b5f5f06-ae75-11e5-8ac6-f04bee4b6189.png)

That's still Gross, but its at least a Fixed-Width value of Gross.

I suspect this is more a "Browser side" bug, but I'm writing this here because I was asked ;)

**Comment by preaction** [comment]
Or a bug in normalize.css... Sheesh that's horrendous. We can work around it though. And bump a note upstream to the normalize.css to see if they've fixed it. `font-family: monospace, monospace;` is kind of weird, so I'm not completely surprised a browser's having trouble with it...

**Comment by preaction** [comment]
It's apparently exactly the way it should be and has been since 2009: https://github.com/necolas/normalize.css/issues/196. It apparently prevents `font-family: monospace` from showing up as smaller than normal text (demonstration of the issue: https://en.wikipedia.org/wiki/User:Davidgothberg/Test59). In the discussion MediaWiki had about this issue (https://en.wikipedia.org/wiki/MediaWiki_talk:Common.css/Archive_11#Teletype_style_fix_for_Chrome), it seems like someone else (Simetrical) came across the same problem you did.

TL;DR: This is a proper clusterfuck. But maybe in the 6 years since then, browsers have advanced enough to make another option viable...

**Comment by kentfredric** [comment]
Its worth mentioning that the `bootstrap` theme doesn't suffer this issue, and everything else looks much more natural with that theme to me than the default theme.

But yeah, with just `monospace`, it does look too small and ugly. :/

**Comment by preaction** [comment]
Huh. The bootstrap theme must not use normalize.css, they must have their own thing. Which means it might be possible to override that particular bit of normalize.css... Or maybe use what the bootstrap one uses...

**Comment by kentfredric** [comment]
Interesting. `bootstrap.css` has *two* CSS rules for the same set of elements.
```css
code,
kbd,
pre,
samp {
  font-family: monospace, monospace;
  font-size: 1em;
}
```
....
```css
code,
kbd,
pre,
samp {
  font-family: Menlo, Monaco, Consolas, "Courier New", monospace;
}
```

Why? Who knows. =)

**Comment by preaction** [comment]
Hah. The first one probably comes from normalize.css, but since bootstrap comes with its own decision as to which font to use, the second one "fixes" this problem ;) Indeed, that'd probably work just fine for the `share/theme/default/css/statocles-default.css` to avoid this terrible, terrible problem.

Does `monospace, sans-serif` fix the problem? Or maybe `monospace, inherit`? If they do, then I'd expect `monospace, serif` to work, but if `monospace, serif` fails, it'll be impossible to tell the difference between "it picked the serif font" and "the font named 'monospace, serif' doesn't exist". And normalize.css switched from `monospace, serif` to `monospace, monospace` for some reason...

But if either of those work, then what's going on? Is Firefox somehow misinterpreting the generic family name when it isn't the last thing?

**Comment by kentfredric** [comment]
code | browser font | browser size | fc-match output
-------|------------------|-----------------|----------------------
monospace, serif | Times New Roman | 13.5px | "Times New Roman" "Regular"
serif, monospace | Times New Roman | 13.5px | "Times New Roman" "Regular"
monospace, inherit | Courier New | 10.13333px | "Courier New" "Regular"
serif | Times New Roman  |13.5px | "Times New Roman" "Regular"
monospace | Courier New | 10.13333px | "Courier New" "Regular"
monospace,monospace | Times New Roman | 13.5px | "Courier New" "Regular"

**Comment by preaction** [comment]
Thanks. Looks like Bootstrap's method will work best then.

**Comment by kentfredric** [comment]
Hah. Also, the Wikipedia test page, the top examples are all in Times New Roman. 

But if I try to get a print preview to make a screenshot via PS printing .... they all become courier new.

**Comment by kentfredric** [comment]
This is how https://en.wikipedia.org/wiki/User:Davidgothberg/Test59 looks for me.

http://imgur.com/a/3tUJv

But strangely, it looks different in the print preview.

http://imgur.com/a/hyMmS

**Comment by preaction** [comment]
If you add that bootstrap block to your `theme/css/statocles-default.css`, does that also fix it? The one that goes:

```
code, kbd, pre, samp {
    font-family: Menlo, Monaco, Consolas, "Courier New", monospace;
}
```

Anywhere should be fine, but I'm likely going to put it at the top. If so, I'll commit that to the repo so future people won't get bit.

**Comment by kentfredric** [comment]
I did in fact attempt injecting: 
```code,
kbd,
pre,
samp {
  font-family: Menlo, Monaco, Consolas, "Courier New", monospace;
}
```
Immediately after the existing block in `theme/css/normalise.css` at one point, and it had the desired effect. But I've now tweaked all my font-config settings and the problem is no-longer reproducible.

So the trick now is to work out *which* font-config setting caused the pain.

**Comment by preaction** [comment]
Okay, so I'll reset my commit adding that block, but we can come back to this if the issue starts spreading. I'm increasingly of the opinion that Skeleton basically requires you to add even more bits on top of it, and I'm starting to think I need to switch to using something else...
